### ISSUE TITLE
Latest bcc package fails with "invalid BPF_LD_IMM64 insn" errors

### ISSUE THREAD

**Issue by brendangregg** [iovisor/bcc#1758]
I went to debug a prod issue but nothing works. Looks like the bcc packages have been updated in our BaseAMI, and now get invalid BPF_LD_IMM64 insn errors:

```
# /usr/share/bcc/tools/opensnoop
bpf: Failed to load program: Unknown error 524
0: (79) r7 = *(u64 *)(r1 +104)
1: (b7) r1 = 0
2: (7b) *(u64 *)(r10 -8) = r1
3: (7b) *(u64 *)(r10 -16) = r1
4: (7b) *(u64 *)(r10 -24) = r1
5: (7b) *(u64 *)(r10 -32) = r1
6: (7b) *(u64 *)(r10 -40) = r1
7: (85) call 14
8: (bf) r6 = r0
9: (7b) *(u64 *)(r10 -48) = r6
10: (bf) r1 = r10
11: (07) r1 += -24
12: (b7) r2 = 16
13: (85) call 16
14: (67) r0 <<= 32
15: (77) r0 >>= 32
16: (55) if r0 != 0x0 goto pc+12
 R0=imm0 R6=inv R7=inv R10=fp
17: (7b) *(u64 *)(r10 -40) = r6
18: (85) call 5
19: (7b) *(u64 *)(r10 -32) = r0
20: (7b) *(u64 *)(r10 -8) = r7
21: (18) r1 = 0xffff880b034ef080
23: (bf) r2 = r10
24: (07) r2 += -48
25: (bf) r3 = r10
26: (07) r3 += -40
27: (b7) r4 = 0
28: (85) call 2
29: (b7) r0 = 0
30: (95) exit

from 16 to 29: safe

Traceback (most recent call last):
  File "/usr/share/bcc/tools/opensnoop", line 138, in <module>
    b.attach_kprobe(event="do_sys_open", fn_name="trace_entry")
  File "/usr/lib/python2.7/dist-packages/bcc/__init__.py", line 569, in attach_kprobe
    fn = self.load_func(fn_name, BPF.KPROBE)
  File "/usr/lib/python2.7/dist-packages/bcc/__init__.py", line 368, in load_func
    (func_name, errstr))
Exception: Failed to load BPF program trace_entry: Unknown error 524

# dmesg | tail
[81841.522606] read cpu 12 ibrs val 0
[81841.522608] read cpu 13 ibrs val 0
[81841.522610] read cpu 14 ibrs val 0
[81841.522611] read cpu 15 ibrs val 0
[81880.803312] invalid BPF_LD_IMM64 insn
[81880.805749] invalid BPF_LD_IMM64 insn
[81880.920113] invalid BPF_LD_IMM64 insn
[81880.922547] invalid BPF_LD_IMM64 insn
[81916.707596] invalid BPF_LD_IMM64 insn
[81916.710032] invalid BPF_LD_IMM64 insn

# dpkg -l | grep bcc
ii  bcc-tools                           0.5.0-364.git.cf44278                      all          Command line tools for BPF Compiler Collection (BCC)
ii  libbcc                              0.5.0-364.git.cf44278                      amd64        Shared Library for BPF Compiler Collection (BCC)
ii  libcc1-0:amd64                      5.4.0-6ubuntu1~16.04.9                     amd64        GCC cc1 plugin for GDB
ii  python-bcc                          0.5.0-364.git.cf44278                      all          Python wrappers for BPF Compiler Collection (BCC)

# uname -a
Linux xxx 4.4.0-122-generic #146-Ubuntu SMP Mon Apr 23 15:34:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

cf44278 is pretty recent: Apr 30. If this issue has already been fixed, I've been unable to find it so far.

**Comment by yonghong-song** [comment]
Typically the error "invalid BPF_LD_IMM64 insn" is due to relocation error. I am not aware of breakage on this recently, but I am using latest kernel and latest llvm.

**Comment by yonghong-song** [comment]
The error 524 indicates ENOTSUPP. Maybe some recent change results in 4.4 verifier unhappy?

**Comment by brendangregg** [comment]
I rolled back 400 commits in bcc, and it still fails in the same way. I'm thinking the problem is this Ubuntu 4.4 kernel, which has been updated since bcc worked last. It's now 4.4.0-122-generic:

```
# uname -a
Linux xxx 4.4.0-122-generic #146-Ubuntu SMP Mon Apr 23 15:34:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
```

Maybe a security patch for the BPF verifier was backported into 4.4 by Canonical?

**Comment by yonghong-song** [comment]
Mostly a ubuntu issue. See https://github.com/iovisor/bcc/issues/1678.
Looks like 4.4.0-122-generic has the issue. You may want to wait/try for 4.4.0-123.

**Comment by brendangregg** [comment]
ok, thanks, I see in that thread that it was fixed in Ubuntu "4.4.0.125-generic". I'll close this issue for now.
